### PR TITLE
nixos: Libinput: Add support like there is for synaptics

### LINKED ISSUE #9920: libinput synaptics like configuration support
Already started here:

https://github.com/eskimor/nixpkgs/tree/libinput-master

It works for me, but not all options are supported yet. I hope to find time polishing it and make a proper pull request soon. In the meantime, you can use what I have, if you need it.

### PR DESCRIPTION
**Pull request by eskimor**
Closes #9920

Works for me, but I have not tried out all of the supported options.

### CODE REVIEW

**Comment by colemickens** [comment]
+1

Also, thanks for fixing my mistakes. I was about to let you know that my changes were wrong (weren't the right bool format), I forgot I was in Wayland when "testing" the change. It will be nice to have this in!

**Comment by colemickens** [comment]
Pinging this, can this be merged?

**Comment by zimbatm** [comment]
Thanks !

**Comment by Ericson2314** [comment]
Ah I somehow missed this. Great work!

**Comment by eskimor** [comment]
Thank you! Glad I could help out. Should any bugs show up - I will fix them, just notify me with @eskimor. (Obviously this will never happen, though ;-) )

**Comment by zimbatm** [comment]
Hehe

File: nixos/modules/services/x11/hardware/libinput.nix [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+
+let cfg = config.services.xserver.libinput;
+    xorgBool = v: if v then "on" else "off";
+in {
+
+  options = {
+
+    services.xserver.libinput = {
+
+      enable = mkOption {
```
Comment (jagajaga) [inline nixos/modules/services/x11/hardware/libinput.nix]: @zimbatm we ask people to use `mkEnableOption`.

**Comment by eskimor** [comment]
@jagajaga PR for that is in the works

File: nixos/modules/services/x11/hardware/libinput.nix (hunk 1 above)
Comment (zimbatm) [inline nixos/modules/services/x11/hardware/libinput.nix]: Thanks, I see that it has been fixed on master. Thanks @eskimor !

Comment (eskimor) [inline nixos/modules/services/x11/hardware/libinput.nix]: :-)

**Pull request merged**